data/helpers.d/package

```diff
@@ -55,48 +71,72 @@ ynh_package_install() {
 #
 # usage: ynh_package_install_from_equivs controlfile
 # | arg: controlfile - path of the equivs control file
-ynh_package_install_from_equivs() {
+ynh_package_install_from_equivs () {
     controlfile=$1
 
-    # install equivs package as needed
+    # Check if the equivs package is installed. Or install it.
     ynh_package_is_installed 'equivs' \
         || ynh_package_install equivs
 
     # retrieve package information
-    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)
-    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)
+    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)	# Retrieve the name of the debian package
+    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)	# And its version number
     [[ -z "$pkgname" || -z "$pkgversion" ]] \
-        && echo "Invalid control file" && exit 1
+        && echo "Invalid control file" && exit 1	# Check if this 2 variables aren't empty.
 
-    # update packages cache
+    # Update packages cache
     ynh_package_update
 
-    # build and install the package
-    TMPDIR=$(ynh_mkdir_tmp)
-    (cp "$controlfile" "${TMPDIR}/control" \
-     && cd "$TMPDIR" \
+    # Build and install the package
+    TMPDIR=$(mktemp -d)
+    # Note that the cd executes into a sub shell
+    # Create a fake deb package with equivs-build and the given control file
+    # Install the fake package without its dependencies with dpkg
+    # Install missing dependencies with ynh_package_install
+    (cp "$controlfile" "${TMPDIR}/control" && cd "$TMPDIR" \
      && equivs-build ./control 1>/dev/null \
      && sudo dpkg --force-depends \
           -i "./${pkgname}_${pkgversion}_all.deb" 2>&1 \
-     && ynh_package_install -f) \
-        && ([[ -n "$TMPDIR" ]] && rm -rf $TMPDIR)
+     && ynh_package_install -f)
+    [[ -n "$TMPDIR" ]] && rm -rf $TMPDIR	# Remove the temp dir.
 
     # check if the package is actually installed
     ynh_package_is_installed "$pkgname"
 }
 
-# Remove package(s)
+# Install dependencies with a equivs control file
 #
-# usage: ynh_package_remove name [name [...]]
-# | arg: name - the package name to remove
-ynh_package_remove() {
-    ynh_apt remove $@
+# usage: ynh_app_dependencies dep [dep [...]]
```

Comment:
Imho this should be renamed as `ynh_install_app_dependencies` to be the symetric of `ynh_remove_app_dependencies`